<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>768</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Particle demo QT SFML</string>
  </property>
  <property name="locale">
   <locale language="English" country="Belgium"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="Canvas" name="SFMLWidget" native="true">
      <property name="minimumSize">
       <size>
        <width>600</width>
        <height>400</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBoxTools">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>200</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Tools</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QWidget" name="widget" native="true">
         <layout class="QFormLayout" name="formLayout">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <item row="0" column="1">
           <widget class="QGroupBox" name="groupBoxAngles">
            <property name="minimumSize">
             <size>
              <width>180</width>
              <height>140</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>0</width>
              <height>100</height>
             </size>
            </property>
            <property name="title">
             <string>Angle</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_3">
               <item>
                <widget class="QSlider" name="sliderRange">
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>360</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_2">
                 <item>
                  <widget class="QLabel" name="labelRange">
                   <property name="text">
                    <string>&amp;Range:</string>
                   </property>
                   <property name="buddy">
                    <cstring>lineEditRange</cstring>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="lineEditRange">
                   <property name="text">
                    <string>0</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_4">
               <item>
                <widget class="QSlider" name="sliderRotation">
                 <property name="maximum">
                  <number>360</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_3">
                 <item>
                  <widget class="QLabel" name="labelRotation">
                   <property name="text">
                    <string>R&amp;otation</string>
                   </property>
                   <property name="buddy">
                    <cstring>lineEditRotation</cstring>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="lineEditRotation">
                   <property name="text">
                    <string>0</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QGroupBox" name="groupBoxParticles">
            <property name="title">
             <string>Particles</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_8">
             <property name="leftMargin">
              <number>9</number>
             </property>
             <property name="rightMargin">
              <number>9</number>
             </property>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayoutColor">
               <item>
                <widget class="QLabel" name="labelColor">
                 <property name="text">
                  <string>&amp;Color:</string>
                 </property>
                 <property name="buddy">
                  <cstring>lineEditColor</cstring>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="lineEditColor"/>
               </item>
               <item>
                <widget class="QToolButton" name="toolButtonColor">
                 <property name="text">
                  <string>...</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayoutAmount">
               <item>
                <widget class="QSlider" name="sliderAmount">
                 <property name="minimum">
                  <number>100</number>
                 </property>
                 <property name="maximum">
                  <number>100000</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_4">
                 <item>
                  <widget class="QLabel" name="labelAmount">
                   <property name="text">
                    <string>&amp;Amount:</string>
                   </property>
                   <property name="buddy">
                    <cstring>lineEditAmount</cstring>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="lineEditAmount">
                   <property name="text">
                    <string>0</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayoutSpeed">
               <item>
                <widget class="QSlider" name="sliderSpeed">
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>1000</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayoutSpeed">
                 <item>
                  <widget class="QLabel" name="labelSpeed">
                   <property name="text">
                    <string>&amp;Speed: </string>
                   </property>
                   <property name="buddy">
                    <cstring>lineEditSpeed</cstring>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="lineEditSpeed">
                   <property name="text">
                    <string>0</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayoutLife">
               <item>
                <widget class="QSlider" name="sliderLife">
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>50</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayoutLife">
                 <item>
                  <widget class="QLabel" name="labelLife">
                   <property name="text">
                    <string>&amp;Lifespan:</string>
                   </property>
                   <property name="buddy">
                    <cstring>lineEditLife</cstring>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="lineEditLife">
                   <property name="inputMask">
                    <string/>
                   </property>
                   <property name="text">
                    <string>0</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionClear"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="text">
    <string>&amp;Clear</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Canvas</class>
   <extends>QWidget</extends>
   <header>canvas.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>sliderRange</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setRange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>572</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>342</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderRotation</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setRotation(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>572</x>
     <y>172</y>
    </hint>
    <hint type="destinationlabel">
     <x>342</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderAmount</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setAmount(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>682</x>
     <y>270</y>
    </hint>
    <hint type="destinationlabel">
     <x>394</x>
     <y>247</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderSpeed</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setSpeed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>682</x>
     <y>323</y>
    </hint>
    <hint type="destinationlabel">
     <x>394</x>
     <y>247</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderLife</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setLife(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>682</x>
     <y>376</y>
    </hint>
    <hint type="destinationlabel">
     <x>394</x>
     <y>247</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setRotation(int)</slot>
  <slot>setAmount(int)</slot>
  <slot>setLife(int)</slot>
  <slot>setRange(int)</slot>
  <slot>setSpeed(int)</slot>
 </slots>
</ui>
